### DISCUSSION TITLE
Custom components by inheriting `ActorComponent`

### DISCUSSION THREAD

**Discussion by prokopst** [nxrighthere/UnrealCLR#50] [Help] [answered]
I've basically started with unreal engine thanks to your plugin. Creating custom components by inheriting `ActorComponent` seems like the way of doing things if I don't want to implement everything in a custom `Actor`. However it's not possible to inherit `ActorComponent` due to its constructor protection levels:
https://github.com/nxrighthere/UnrealCLR/blob/7372bdceef77818c199ee1a8fc606b3db871fd40/Source/Managed/Framework/Framework.cs#L7488
https://github.com/nxrighthere/UnrealCLR/blob/7372bdceef77818c199ee1a8fc606b3db871fd40/Source/Managed/Framework/Framework.cs#L7490

If I understand correctly sounds like the right constructor hasn't been implemented/exposed yet.

If it's not a rocket science I wouldn't mind to do that on my own, but I would appreciate some pointers.

**Comment by nxrighthere** [comment, marked as answer]
You are trying to create a class derived from a non-instantiable abstract `ActorComponent` which can't be used for components in managed code since it has no native representation in the world from the engine's perspective:

![ч](https://user-images.githubusercontent.com/8279330/96420439-f1c51800-120e-11eb-89ce-305d676ecf02.PNG)

Use `SceneComponent` with classes that derived from it.
